connector/connect/common/src/main/protobuf/spark/connect/base.proto

```diff
@@ -199,6 +200,17 @@ message AnalyzePlanRequest {
     // (Required) The logical plan to get the storage level.
     Relation relation = 1;
   }
+
+  message Checkpoint {
```

Comment:
Ok, can we have a single mechanism of invoking checkpoints, not both Analyze and Execute. I prefer moving them to Execute given the (potentially) long running nature of an eager checkpoint. Please keep in mind that this might require some specialized handling on the client side.

Reply:
Yeah that works to me.

Reply:
Actually taking a look at `ExecutePlanRequest`/`ExecutePlanResponse`, I think it's better to keep it in `AnalyzePlanRequest`. `ExecutePlanResponse` assumes that the output is actual output values such as Arrow batch, map, etc. vs `AnalyzePlanResponse` can assume it's a DataFrame. This is also consistent with `persist`/`cache`.

I think it's better to take `RemoveCachedRemoteRelationCommand` as a separate command that is not dedicated to `checkpoint`. It can be used in other places if we happen to reuse `CachedRemoteRelationCommand` somewhere.

Reply:
Caching is lazy, so it cannot trigger execution of the main query. Checkpoint can actually do this, which is weird for an analyze request. Moreover analyse does not work well with long running operations (i.e. an eager checkpoint) for the following reasons::
- Analyze needs a thread in the server's thread pool which in extreme cases can lead to exhaustion, or in a less extreme scenario can lead to reduced availability.
- Long running request can be killed by intermediate proxies. Analyze does not support reattach which is an issue when the checkpoint takes too long. In that case you are basically dead in the water.

The argument that ExecutePlanResponse does not support this is IMO a bit flimsy. Look for example at the `SqlCommandResult` this can be returned as part of the `ExecutePlanResponse` and it returns a relation.